src/common/helpers/invoices/invoice-sum-inclusive.ts

```diff
@@ -0,0 +1,249 @@
+import { InvoiceItemSumInclusive } from './invoice-item-sum-inclusive';
+import { Invoice } from 'common/interfaces/invoice';
+import collect from 'collect.js';
+import { InvoiceStatus } from 'common/enums/invoice-status';
+
+export class InvoiceSumInclusive {
+  protected taxMap = collect();
+  protected invoiceItems = new InvoiceItemSumInclusive(this.invoice);
+  protected totalTaxMap: Record<string, unknown>[] = [];
+
+  public totalDiscount = 0;
+  public total = 0;
+  public totalTaxes = 0;
+  public totalCustomValues = 0;
+  public subTotal = 0;
+
+  constructor(public invoice: Invoice) {}
+
+  public async build() {
+    await this.calculateLineItems();
+    await this.calculateDiscount();
+    await this.calculateCustomValues();
+    await this.calculateInvoiceTaxes();
+    await this.setTaxMap();
+    await this.calculateTotals();
+    await this.calculateBalance();
+    await this.calculatePartial();
+
+    return this;
+  }
+
+  protected async calculateLineItems() {
+
+    await this.invoiceItems.process();
+
+    this.invoice.line_items = this.invoiceItems.lineItems;
+    this.total = this.invoiceItems.subTotal;
+    this.subTotal = this.invoiceItems.subTotal;
+    
+    return this;
+  }
+
+  protected async calculateDiscount() {
+    this.totalDiscount = this.discount(this.invoiceItems.subTotal);
+    this.total -= this.totalDiscount;
+
+    return this;
+  }
+
+  protected async calculateInvoiceTaxes() {
+
+    let amount = this.total;
```

Comment:
This variable isn't used anywhere. Should we return it or update some state based on it, or?

Reply:
pushed changes for this. needed to actually use the var :)